newrelic-agent/src/main/java/com/newrelic/agent/Agent.java

```diff
@@ -76,6 +76,11 @@ public static boolean isDebugEnabled() {
         return DEBUG;
     }
 
+    private static boolean setDebug(){
```

Comment:
@meiao  Are you thinking to include the other static members in the block or just for DEBUG?

Or just:
```
private static final boolean DEBUG;

    static{
        DEBUG = setDebug();
        //OR use below and drop the method?
        //DEBUG =Boolean.getBoolean("newrelic.debug") || Boolean.parseBoolean(System.getenv("NEWRELIC_DEBUG"));
    }

```

`isDebugEnabled()` is public so other places can already check. Do we want other places to be able to set?  It was actually a final variable, so should only set once...unless you think change that?

Reply:
I just added setDebug as a private member because I thought it might look cleaner to move the boolean checks out of that section since there are multiple checks.

Reply:
My first idea was to drop the method.
But as I read thru the PR, there are 3 other places that make the same logic:
```
Boolean.getBoolean(newrelicDebug) || Boolean.parseBoolean(System.getenv("NEWRELIC_DEBUG"));
```
So instead of copying that all the time, it could be a method that all call.

Reply:
I see. One of the problems with one of the places is that Agent class hasn't been initialized when it is called.   It was unclear to me why the original author didn't decide to make static call from AgentJarHelper where they had to duplicate it.  
https://github.com/newrelic/newrelic-java-agent/blob/0638d8646910e45c84ec3a11e7507947a57979af/newrelic-agent/src/main/java/com/newrelic/agent/config/AgentJarHelper.java#L226

Do you think leave AgentJarHelper as is and just the Agent class in the other places where we can? Or something else?

Reply:
OH, using it in AgentJarHelper would initialize it wouldn't it.